[webkitpy] Apply timeouts to individual simctl commands

JonWBedard opened the pull request:
#### 7e02e7983e144a94ed6697eacc920204db4ddae2
<pre>
[webkitpy] Apply timeouts to individual simctl commands
<a href="https://bugs.webkit.org/show_bug.cgi?id=302826">https://bugs.webkit.org/show_bug.cgi?id=302826</a>
<a href="https://rdar.apple.com/165085170">rdar://165085170</a>

Reviewed by Elliott Williams.

There are some simctl commands which are hanging on iOS 26. We haven&apos;t deeply
investigated to figure out which ones, although I have some suspicions. Guard
the simctl commands most likely to hang with timeouts.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager._boot_device):
(SimulatedDevice.is_usable):

Canonical link: <a href="https://commits.webkit.org/303304@main">https://commits.webkit.org/303304@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/c6855e8bafb1f1d6d824763d9cf9692ccb6fda3c

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/131999 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/4491 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/43015 "Built successfully") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/139513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/83903 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/133869 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/4434 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/4253 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/139513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/83903 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 mac-apple 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/134945 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/3160 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/118220 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/139513 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | ⏳ 🛠 vision-apple 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/131348 "Passed tests") | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/3054 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac-debug~~](https://ews-build.webkit.org/#/builders/165/builds/921 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/82731 "Built successfully") | | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/111808 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/36339 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/142158 "Built successfully") | | 
| | [  ~~🛠 vision~~](https://ews-build.webkit.org/#/builders/153/builds/4161 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/136/builds/36922 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/109272 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [  ~~🛠 vision-sim~~](https://ews-build.webkit.org/#/builders/160/builds/4242 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/161/builds/3629 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/109444 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-26-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/114498 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/57382 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/20521 "Built successfully and passed tests") | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/158/builds/4214 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/32894 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/157/builds/4046 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/4306 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/4174 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

bls1999 commented on Tools/Scripts/webkitpy/xcode/simulated_device.py:
I feel like this would be better as its own method? Like maybe `run_command_with_timeout`? Or maybe something more bespoke for `simctl` commands like `_invoke_simctl`.

JonWBedard replied:
`run_command` already supports considering timeouts, which is why `with Timeout(..., patch=False)` works, and we already use that elsewhere in this same file

bls1999 replied:
To clarify, my reasoning for the suggestion is that there's a bunch of these individual `simctl` calls specifically in `simulated_device.py`. If we're moving towards adding a bunch of timeouts for `simctl` when we already have some as it is, then we may just want to change all `simctl` invocations to go through one function that supports timeouts.

Either way, it's not a huge enough thing to block on. So [unofficial] r+ from me.

emw-apple reviewed (approved):
15 and 5 seconds to boot and launch, respectively, seem a little fast to me. Maybe you raise these to something more exceptional (60 sec?) that is still fast enough to not affect infra throughput.

bls1999 reviewed (changes requested):
It just occurred to me that, under load, the machines we have in our infrastructure could take up to 5 minutes to boot individual simulators. I think we need to bump the timeout way up and only use it as a sanity check (maybe 10 minutes?). We shouldn't hit that under normal circumstances.

JonWBedard committed:
[webkitpy] Apply timeouts to individual simctl commands
https://bugs.webkit.org/show_bug.cgi?id=302826
rdar://165085170

Reviewed by Elliott Williams.

There are some simctl commands which are hanging on iOS 26. We haven't deeply
investigated to figure out which ones, although I have some suspicions. Guard
the simctl commands most likely to hang with timeouts.

* Tools/Scripts/webkitpy/xcode/simulated_device.py:
(SimulatedDeviceManager._boot_device):
(SimulatedDevice.is_usable):

Canonical link: https://commits.webkit.org/303304@main
